src/nvim/normal.c

```diff
@@ -3386,8 +3386,8 @@ bool add_to_showcmd(int c)
 
 void add_to_showcmd_c(int c)
 {
-  if (!add_to_showcmd(c))
-    setcursor();
+  add_to_showcmd(c);
+  setcursor();
```

Comment:
@bfredl 
I assume this also affects Vim then?

Reply:
What affects vim? This is just a consequence of the other change (display_showcmd neither moves cursor nor restores it) in this commit.

Reply:
Since this is also code used in Vim I've thought it might be relevant for it also (to avoid flicker), i.e. it could have come into Neovim through a Vim patch also.

Reply:
The original issue was reported to occur in nvim only but not vim.  The relevant abstraction that this patch makes use of (separate input and drawing cursors internally) does not exist in vim, so I'm not sure what the vim patch would look like.

Reply:
Ok, thanks for the clarification!